modules/git/repo_commit.go

```diff
@@ -203,35 +203,52 @@ func (repo *Repository) commitsByRange(id SHA1, page int) (*list.List, error) {
 
 func (repo *Repository) searchCommits(id SHA1, opts SearchCommitsOptions) (*list.List, error) {
 	cmd := NewCommand("log", id.String(), "-100", "-i", prettyLogFormat)
-	if len(opts.Keywords) > 0 {
-		for _, v := range opts.Keywords {
-			cmd.AddArguments("--grep=" + v)
-		}
-	}
+	var args []string
 	if len(opts.Authors) > 0 {
 		for _, v := range opts.Authors {
-			cmd.AddArguments("--author=" + v)
+			args = append(args, "--author="+v)
 		}
 	}
 	if len(opts.Committers) > 0 {
 		for _, v := range opts.Committers {
-			cmd.AddArguments("--committer=" + v)
+			args = append(args, "--committer="+v)
 		}
 	}
 	if len(opts.After) > 0 {
-		cmd.AddArguments("--after=" + opts.After)
+		args = append(args, "--after="+opts.After)
 	}
 	if len(opts.Before) > 0 {
-		cmd.AddArguments("--before=" + opts.Before)
+		args = append(args, "--before="+opts.Before)
 	}
 	if opts.All {
-		cmd.AddArguments("--all")
+		args = append(args, "--all")
+	}
+	if len(opts.Keywords) > 0 {
+		for _, v := range opts.Keywords {
+			cmd.AddArguments("--grep=" + v)
+		}
 	}
+	cmd.AddArguments(args...)
 	stdout, err := cmd.RunInDirBytes(repo.Path)
 	if err != nil {
 		return nil, err
 	}
-	return repo.parsePrettyFormatLogToList(stdout)
+	if len(opts.Keywords) > 0 {
+		for _, v := range opts.Keywords {
+			if len(v) >= 6 {
```

Comment:
The constant 6 here is a little bit mysterious.

Reply:
My bad, I kind of changed that without thinking about it. Changed it back to 4. The idea is it's probably not worth checking strings that are too short.

Reply:
I think the constant 4 or 6 deserves a comment. I can figure that it is related to how `git` recognize the shortest allowed input hash. But it is not apparent simply looking at the code.